src/components/Home.jsx

```diff
@@ -1,19 +1,97 @@
-import React from 'react';
+import React, { useState } from 'react';
 import getToken from '../lib/tokens';
 import { useHistory } from 'react-router-dom';
+import db from '../lib/firebase';
+import firebase from 'firebase/app';
+
+import './Home.css';
 
 const Home = () => {
   let history = useHistory();
+  const [tokenName, setTokenName] = useState('');
+  const [notification, setNotification] = useState(null);
 
   const generateToken = () => {
     const token = getToken();
     localStorage.setItem('token', token);
+
+    db.collection('tokens')
+      .doc('tokenDocument')
+      .update({
+        tokenArray: firebase.firestore.FieldValue.arrayUnion(token),
+      })
+      .catch((error) => {
+        db.collection('tokens')
+          .doc('tokenDocument')
+          .set({
+            tokenArray: [token],
+          });
+      });
+
     history.push('/list-view');
   };
 
+  const resetError = () => {
+    setTimeout(() => {
+      setNotification(null);
+    }, 10000);
+  };
+
+  const compareToken = (e) => {
+    e.preventDefault();
+
+    db.collection('tokens')
+      .where('tokenArray', 'array-contains', tokenName)
```

Comment:
Because we are now storing the token in a collection called 'tokens', we can query the array they're added to in this way and it will return all documents where the array has a matching value. It seemed like the most straightforward query as it just checks through every array value for a match with the token the user entered ☺️
The '==' query that you and Zachee used obviously works great for what you're doing, as you just need to search for a specific value in a document rather than search in an array.

Reply:
I think querying `items` collection and finding a matching `token` can be a sufficient operation. It can help us to avoid creating a new array of token names that checks against all the documents in the DB. 

Is there a particular reason why you decided to take this route?